client/src/js/services/grid/GridGrouping.js

```diff
@@ -129,31 +138,88 @@ function GridGroupingService(uiGridGroupingConstants, $filter, Session, $timeout
    */
   function configureDefaultGroupingOptions(gridApi) {
 
-    // bind the group selection method
-    gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+    //this instruction block can be executed if the grid involves selection functionality
+    if(gridApi.selection){
+
+      // bind the group selection method
+      gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+
+      gridApi.selection.on.rowSelectionChangedBatch(null, handleBatchSelection.bind(this));
+
+    }
 
     // hook into rows rendered call to ensure the grid is ready before expanding initial nodes
     gridApi.core.on.rowsRendered(null, util.once(function () {
-      gridApi.grouping.groupColumn('trans_id');
+      gridApi.grouping.groupColumn(this.column);
 
       // for the expandAllRows() to be fired last
-      $timeout(gridApi.treeBase.expandAllRows, 0, false);
-    }));
+      unfoldAll(gridApi);
```

Comment:
I would add `Groups` to this name: `unfoldAllGroups(api)`.

It would also be nice if you didn't have to pass in `gridApi` since none of the other functions do that.  It will keep your code uniform.